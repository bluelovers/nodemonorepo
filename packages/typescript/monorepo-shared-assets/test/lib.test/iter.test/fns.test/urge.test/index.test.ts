import {iter} from '../../../../../index'
import urge = iter.fns.urge

const iterable = 'abcdefghijklmno'
const getArray = (fn?: urge.RemainingHandler<string>) => Array.from(urge(iterable, 4, fn))

it('keep remaining part', () => {
  const array = getArray(urge.KEEP_REMAINING_PART)
  expect(array).toEqual(getArray()) // default parameter
  expect(array).toMatchSnapshot()
  expect(array.length).toBe(4)
  expect(array[3]).toEqual([...'mno'])
})

it('omit remaining part', () => {
  const array = getArray(urge.OMIT_REMAINING_PART)
  expect(array).toMatchSnapshot()
  expect(array.length).toBe(3)
})
