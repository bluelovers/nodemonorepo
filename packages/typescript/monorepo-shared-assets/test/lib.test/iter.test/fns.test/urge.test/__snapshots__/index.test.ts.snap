// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keep remaining part 1`] = `
Array [
  Array [
    "a",
    "b",
    "c",
    "d",
  ],
  Array [
    "e",
    "f",
    "g",
    "h",
  ],
  Array [
    "i",
    "j",
    "k",
    "l",
  ],
  Array [
    "m",
    "n",
    "o",
  ],
]
`;

exports[`omit remaining part 1`] = `
Array [
  Array [
    "a",
    "b",
    "c",
    "d",
  ],
  Array [
    "e",
    "f",
    "g",
    "h",
  ],
  Array [
    "i",
    "j",
    "k",
    "l",
  ],
]
`;
